tests/test_schemas.py

```diff
@@ -84,61 +71,10 @@ def getInvalidValue(self, cls, fieldName):
         Returns a value that should trigger a schema validation failure.
         """
         field = self.getAvroSchema(cls, fieldName)
-        typ = field.type
-        if isinstance(typ, avro.schema.UnionSchema):
-            typ = typ.schemas[1]
-        if isinstance(typ, avro.schema.MapSchema):
-            ret = ["string"]
-        elif isinstance(typ, avro.schema.ArraySchema):
-            ret = 42.0
-        elif isinstance(typ, avro.schema.EnumSchema):
-            ret = "NO ENUM COULD HAVE THIS NAME"
-        elif isinstance(typ, avro.schema.RecordSchema):
-            ret = 42
-        elif typ.type in self.badValueMap:
-            ret = self.badValueMap[typ.type]
-        else:
-            raise Exception("schema assumptions violated")
-        return ret
-
-    def getRandomValue(self, cls, fieldName):
-        """
-        Returns a value randomly generated conforming to the schema.
-        """
-        maxListSize = 10
-        field = self.getAvroSchema(cls, fieldName)
-        typ = field.type
-        if isinstance(typ, avro.schema.UnionSchema):
-            # TODO put in some probability that return None if
-            # we have a union schema
-            typ = typ.schemas[1]
-        if isinstance(typ, avro.schema.MapSchema):
-            ret = {}
-            for j in range(random.randint(0, maxListSize)):
-                randStr = [randomString() for _ in range(
-                    random.randint(0, maxListSize))]
-                ret["key{0}".format(j)] = randStr
-        elif isinstance(typ, avro.schema.ArraySchema):
-            randInt = random.randint(0, maxListSize)
-            if cls.isEmbeddedType(field.name):
-                embeddedClass = cls.getEmbeddedType(field.name)
-
-                def getRandInst():
-                    return self.getRandomInstance(embeddedClass)
-            else:
-                getRandInst = self.randomValueMap[typ.items.type]
-            ret = [getRandInst() for j in range(randInt)]
-        elif isinstance(typ, avro.schema.EnumSchema):
-            ret = random.choice(typ.symbols)
-        elif isinstance(typ, avro.schema.RecordSchema):
-            self.assertTrue(cls.isEmbeddedType(fieldName))
-            embeddedClass = cls.getEmbeddedType(fieldName)
-            ret = self.getRandomInstance(embeddedClass)
-        elif typ.type in self.randomValueMap:
-            ret = self.randomValueMap[typ.type]()
-        else:
-            raise Exception("schema assumptions violated")
-        return ret
+        t = field.type
+        if isinstance(t, avro.schema.UnionSchema):
+            t = t.schemas[1]
```

Comment:
Nit: this looks like it could be accidentally correct, but is probably good enough?
(e.g. starting with a union (Int, String), we'll try to generate an incorrect int, so we generate random other type, and our RNG selects a string?)